[[ Bug 22663 ]] Support max width/height in Android mobilePickPhoto

montegoulding opened the pull request:
This patch adds support for the max width and height parameters for
`mobilePickPhoto` on android. A previous patch added some support, however, the
behavior was quite different to iOS and the documentation was not updated.

Closes https://quality.livecode.com/show_bug.cgi?id=22663

runrevmark commented on engine/src/java/com/runrev/android/Engine.java:
```diff
@@ -2205,33 +2206,98 @@ else if (resultCode == Activity.RESULT_OK)
+					}
+		
+					Bitmap t_scaled_bitmap = Bitmap.createBitmap (t_bitmap, 0, 0, (int)t_width, (int)t_height, t_matrix, true);
+					t_scaled_bitmap.compress(Bitmap.CompressFormat.JPEG, 100, t_out);
```
So this uses a quality of 100% - can you try and find out what 'quality' level is used on iOS with the means we use there to get a JPEG out of a scaled bitmap?

montegoulding replied:
On iOS it uses `MCjpegquality / 100.0f`, however, `MCjpegquality` appears to have nothing setting it to anything other than 100.

runrevmark replied:
Okay so we should be passing MCjpegquality into the android method then :)

montegoulding replied:
Aha... sorry my mistake I didn't search for it correctly. It's a global prop I didn't know about! OK Will see if this can be used somehow.

runrevmark commented on engine/src/java/com/runrev/android/Engine.java:
```diff
@@ -2205,33 +2206,98 @@ else if (resultCode == Activity.RESULT_OK)
+							break;
+					}
+		
+					Bitmap t_scaled_bitmap = Bitmap.createBitmap (t_bitmap, 0, 0, (int)t_width, (int)t_height, t_matrix, true);
```
Extraneous space after `createBitmap`

montegoulding committed:
[[ Bug 22663 ]] Support max width/height in Android mobilePickPhoto

This patch adds support for the max width and height parameters for
`mobilePickPhoto` on android. A previous patch added some support, however, the
behavior was quite different to iOS and the documentation was not updated.

runrevmark commented:
@livecode-vulcan review ok 28bf1ea

livecode-vulcan commented:
💙 review by @runrevmark ok 28bf1eae3bae93420d9d4902dda3828c5486444b

livecode-vulcan commented:
😎 test success 28bf1eae3bae93420d9d4902dda3828c5486444b

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/763)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/172)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/172)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/172)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/172)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2289)
* try-community-universal-ios-iphoneos13.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.4/builds/23)
* try-community-universal-ios-iphonesimulator13.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.4/builds/23)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2318)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2305)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2304)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2339)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2279)
